daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -1084,8 +1078,10 @@ object Main extends Logging {
         }.map{ _.withExternalVariables(retrieveExternalVariables(performanceOpts.vars, cfgFileNode)) }
          .map{ _.withValidationMode(validate) }
 
-        val rc = processor match {
-          case Some(processor: DataProcessor) if (!processor.isError) => {
+        val rc: ExitCode.Value = processor match {
+          case Some(proc) if (proc.isError) => ExitCode.UnableToCreateProcessor
```

Comment:
Same as above, this case can be removed. We should an the case Some below to match the above code.